app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -4133,4 +4134,185 @@ class PatientRepositoryAndroidTest {
         )
         .inOrder()
   }
+
+  @Test
+  fun searching_for_a_patient_by_numeric_criteria_must_return_list_of_patient_search_results_paritioned_by_assigned_and_not_assigned_facility() {
+    fun createPatientWithNumericCriteriaAndAssignedFacilityID(
+        patientUuid: UUID,
+        phoneNumber: String?,
+        assignedFacilityId: UUID?,
+        businessId: BusinessId?
+    ) {
+      val patientProfile = TestData
+          .patientProfile(
+              patientUuid = patientUuid,
+              generatePhoneNumber = false,
+              patientPhoneNumber = phoneNumber,
+              patientAssignedFacilityId = assignedFacilityId,
+              businessId = businessId
+          )
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+    }
+
+    // given
+    val currentFacility = TestData.facility(uuid = UUID.fromString("c0056c11-105d-4079-80fa-c745128f7fc5"), name = "CHC Bucho")
+    val facility2 = TestData.facility(uuid = UUID.fromString("532264e6-b358-445e-8b48-c6253677db24"), name = "Test")
```

Comment:
Please try to follow consistency when writing tests. This means that if you are naming a facility `currentFacility`, give it an appropriate name like `otherFacility` instead of `facility2`. The same goes for the test data itself, "CHC Bucho" and "Test" do not go well together.